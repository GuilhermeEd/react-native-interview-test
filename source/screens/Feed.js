import React, { Component } from "react";
import { Text, View, FlatList, ActivityIndicator } from "react-native";
import MovieRow from './MovieRow';
import styles from './styles';

export default class Feed extends Component {
  constructor(props) {
    super(props);
  }

  FlatListItemSeparator() {
    return (
      <View style={styles.spacedLine} />
    );
  }

  renderItem = ({ item }) => {
    // Some movies contain null data, so don't render them
    if (!item.title) {
      return;
    }
    return (
      <MovieRow movie={item} likeMovie={this.props.likeMovie} />
    )
  }

  render() {
    const {
      loading,
      data,
    } = this.props;

    if (loading) {
      return (
        <View>
          <Text style={styles.centerText}>LOADING THE FEED</Text>
          <ActivityIndicator
            color='blue'
            size='large'
            style={styles.activityIndicator}
          />
        </View>
      )
    }

    return (
      <View style={styles.flatListContainer}>
        <FlatList
          data={data}
          renderItem={this.renderItem}
          ItemSeparatorComponent={this.FlatListItemSeparator}
          keyExtractor={item => item._id.$oid}
        />
      </View>
    )
  }
}


