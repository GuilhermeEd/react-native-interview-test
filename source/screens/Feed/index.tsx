import React from 'react';
import { Text } from 'react-native';
import { NavigationScreenComponent } from 'react-navigation';

import { Container } from './style';

const Feed: NavigationScreenComponent = () => {
  return (
    <Container testID="feed-screen">
      <Text>Feed</Text>
    </Container>
  );
};

export default Feed;
