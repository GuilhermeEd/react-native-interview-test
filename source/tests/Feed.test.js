import React from 'react';
import renderer from 'react-test-renderer';
import Feed from "../screens/Feed";

const dataMock = [{
    "_id": {
        "$oid": "5b8701a1fc13ae6569000000"
    },
    "title": "Long Live Death (Viva la muerte)",
    "price": 28.704,
    "inventory": 4,
}, {
    "_id": {
        "$oid": "5b8701a1fc13ae6569000001"
    },
    "title": "Home",
    "price": 10.622,
    "inventory": 3,
}];

test('<Feed> renders correctly when loading', () => {
    const tree = renderer.create(<Feed loading={true} data={[]} />).toJSON();
    expect(tree).toMatchSnapshot();
});

test('<Feed> renders correctly when loaded', () => {
    const tree = renderer.create(<Feed loading={false} data={dataMock} />).toJSON();
    expect(tree).toMatchSnapshot();
});