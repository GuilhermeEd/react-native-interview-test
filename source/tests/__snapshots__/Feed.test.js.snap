// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Feed> renders correctly when loaded 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#e8e9ea",
      "borderRadius": 10,
      "height": "85%",
      "padding": 10,
      "width": "100%",
    }
  }
>
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    data={
      Array [
        Object {
          "_id": Object {
            "$oid": "5b8701a1fc13ae6569000000",
          },
          "inventory": 4,
          "price": 28.704,
          "title": "Long Live Death (Viva la muerte)",
        },
        Object {
          "_id": Object {
            "$oid": "5b8701a1fc13ae6569000001",
          },
          "inventory": 3,
          "price": 10.622,
          "title": "Home",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    invertStickyHeaders={undefined}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 20,
                "textAlign": "center",
              }
            }
          >
            Long Live Death (Viva la muerte)
          </Text>
          <View
            className="buttonContainer"
            style={
              Object {
                "display": "flex",
                "flex": 1,
                "flexDirection": "row",
                "height": 60,
                "justifyContent": "space-around",
              }
            }
          >
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "backgroundColor": "#cbe6e6",
                    "borderRadius": 5,
                    "flex": 1,
                    "justifyContent": "center",
                    "margin": 10,
                    "padding": 15,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                >
                  Like
                </Text>
              </View>
            </View>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "backgroundColor": "#cbe6e6",
                    "borderRadius": 5,
                    "flex": 1,
                    "justifyContent": "center",
                    "margin": 10,
                    "padding": 15,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                >
                  View Details
                </Text>
              </View>
            </View>
          </View>
          <View
            className="details"
            style={
              Array [
                Object {
                  "height": 0,
                  "overflow": "hidden",
                  "paddingLeft": 10,
                },
                false,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              Tickets available: 
              4
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              Ticket price: $
              28.704
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#000",
              "height": 0.5,
              "marginBottom": 1,
              "marginTop": 1,
              "width": "100%",
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 20,
                "textAlign": "center",
              }
            }
          >
            Home
          </Text>
          <View
            className="buttonContainer"
            style={
              Object {
                "display": "flex",
                "flex": 1,
                "flexDirection": "row",
                "height": 60,
                "justifyContent": "space-around",
              }
            }
          >
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "backgroundColor": "#cbe6e6",
                    "borderRadius": 5,
                    "flex": 1,
                    "justifyContent": "center",
                    "margin": 10,
                    "padding": 15,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                >
                  Like
                </Text>
              </View>
            </View>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "backgroundColor": "#cbe6e6",
                    "borderRadius": 5,
                    "flex": 1,
                    "justifyContent": "center",
                    "margin": 10,
                    "padding": 15,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                >
                  View Details
                </Text>
              </View>
            </View>
          </View>
          <View
            className="details"
            style={
              Array [
                Object {
                  "height": 0,
                  "overflow": "hidden",
                  "paddingLeft": 10,
                },
                false,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              Tickets available: 
              3
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              Ticket price: $
              10.622
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`<Feed> renders correctly when loading 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    LOADING THE FEED
  </Text>
  <ActivityIndicator
    animating={true}
    color="blue"
    hidesWhenStopped={true}
    size="large"
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "height": 80,
        "justifyContent": "center",
        "margin": 30,
      }
    }
  />
</View>
`;
